frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.jsx

```diff
@@ -71,125 +66,131 @@ const ScalarPeriod = ({ lines = 2, period, onClick }) => (
+        <VariationDetails>{fittedDetailDisplay}</VariationDetails>
+      </VariationContainer>
+    </Tooltip>
+  );
+}
+
 export function SmartScalar({
   onVisualizationClick,
   isDashboard,
   settings,
   visualizationIsClickable,
-  series: [
-    {
-      data: { rows, cols },
-    },
-  ],
+  series,
   rawSeries,
   gridSize,
   width,
   height,
   totalNumGridCols,
   fontFamily,
 }) {
+  const fontSize = "0.875rem";
   const scalarRef = useRef(null);
-  const innerHeight = isDashboard ? height - DASHCARD_HEADER_HEIGHT : height;
 
-  const metricIndex = cols.findIndex(
-    col => col.name === settings["scalar.field"],
+  const insights = rawSeries?.[0].data?.insights;
+  const trend = useMemo(
+    () => computeTrend(series, insights, settings),
+    [series, insights, settings],
   );
-  const dimensionIndex = cols.findIndex(col => isDate(col));
-
-  const lastRow = rows[rows.length - 1];
-  const value = lastRow && lastRow[metricIndex];
-  const column = cols[metricIndex];
-
-  const insights = rawSeries && rawSeries[0].data && rawSeries[0].data.insights;
-  const insight = _.findWhere(insights, { col: column.name });
-  if (!insight) {
+  if (trend == null) {
     return null;
   }
-  const { unit } = insight;
-  const lastDate = lastRow[dimensionIndex];
-  const lastPeriod = formatDateTimeRangeWithUnit([lastDate], unit, {
-    compact: true,
-  });
-
-  const lastValue = insight["last-value"];
-  const formatOptions = settings.column(column);
-
-  const { displayValue, fullScalarValue } = compactifyValue(
-    lastValue,
-    width,
-    formatOptions,
-  );
+  const { value, clicked, comparison, display, formatOptions } = trend;
 
-  const granularity = Lib.describeTemporalUnit(insight["unit"]).toLowerCase();
-
-  const lastChange = insight["last-change"];
-  const previousValue = insight["previous-value"];
-
-  const isNegative = lastChange < 0;
-  const isSwapped = settings["scalar.switch_positive_negative"];
-
-  // if the number is negative but thats been identified as a good thing (e.g. decreased latency somehow?)
-  const changeColor = (isSwapped ? !isNegative : isNegative)
-    ? color("error")
-    : color("success");
-
-  const isPeriodVisible = getIsPeriodVisible(innerHeight);
-  const valueHeight = getValueHeight(innerHeight);
-
-  const changeDisplay = formatChangeAutoPrecision(lastChange, {
-    fontFamily,
-    fontWeight: 900,
-    width: getChangeWidth(width),
-  });
-
-  const tooltipSeparator = <Separator>•</Separator>;
-  const previousValueSeparator = (
-    <PreviousValueSeparator>•</PreviousValueSeparator>
-  );
-  const granularityDisplay = jt`previous ${granularity}`;
-  const previousValueDisplay = (
-    <PreviousValueLabel>
-      {formatValue(previousValue, formatOptions)}
-    </PreviousValueLabel>
-  );
-  const previousValueDisplayCompact = (
-    <PreviousValueLabel>
-      {formatValue(previousValue, {
-        ...formatOptions,
-        compact: true,
-      })}
-    </PreviousValueLabel>
-  );
```

Comment:
I did my best to unify a lot of our elements between the card and tooltip to deduplicate this kind of work.  All work related to comparison display is cleaned up and moved  to `PreviousValueComparison`.